File: Python3/프로그래머스/3/43162. 네트워크/네트워크.py
from collections import deque

def solution(n, computers):
    answer = 0
    visited = [0 for _ in range(n)]
    stack = deque()
    
    def dfs(graph, visited, start):
        stack.append(start)
        while stack:
            j = stack.pop()
            if visited[j] == 0:
                visited[j] = 1
                for k in range(len(graph)):
                    if graph[j][k] == 1 and visited[k] == 0:
                        stack.append(k)
    
    for i in range(len(computers)):
        if visited[i] == 0:
            dfs(computers, visited, i)
            answer += 1
            
    return answer

File: C++/프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

void dfs(int i, vector<int>& visited, const vector<vector<int>>& computers) {
    visited[i] = 1;
    
    for (int j = 0; j < computers[i].size(); ++j) {
        if (computers[i][j] == 1 && visited[j] == 0) {
            dfs(j, visited, computers);
        }
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    vector<int> visited(n, 0);
    
    for (int i = 0; i < n; ++i) {
        if (visited[i] == 0) {
            dfs(i, visited, computers);
            answer++;
        }
    }
    
    return answer;
}